pkg/kn/plugin/context.go

```diff
@@ -0,0 +1,59 @@
+// Copyright © 2023 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+var ctxManager *ContextDataManager
+
+type ContextDataManager struct {
+	ContextData ContextData
+}
+
+func NewContextManager() (*ContextDataManager, error) {
+	if ctxManager == nil {
+		var err error
+		//file, err := os.Open("~/.config/kn/context-cache.json")
+		//if err != nil {
+		//	return nil, err
+		//}
+
+		mockData := `
+{  
+  "service":"hello",
+  "namespace":"default"
+}
```

Comment:
This represents very minimal shared context between every Consumer and Producer to read/write. E.g. in a seems floating value changed on latest usage of `kn` or plugins.

Reply:
This could be further split into plugin specific sub-context and shared sub-context. Where plugin specific is only accessed by it's owner. In addition with read/write permission field maybe?